flang/lib/Lower/Runtime.cpp

```diff
@@ -177,3 +180,42 @@ void Fortran::lower::genPauseStatement(
   auto callee = genRuntimeFunction<mkRTKey(PauseStatement)>(loc, bldr);
   bldr.create<fir::CallOp>(loc, callee, llvm::None);
 }
+
+void Fortran::lower::genDateAndTime(Fortran::lower::FirOpBuilder &builder,
+                                    mlir::Location loc,
+                                    llvm::Optional<fir::CharBoxValue> date,
+                                    llvm::Optional<fir::CharBoxValue> time,
+                                    llvm::Optional<fir::CharBoxValue> zone) {
+  auto callee = genRuntimeFunction<mkRTKey(DateAndTime)>(loc, builder);
+  mlir::Type idxTy = builder.getIndexType();
+  mlir::Value zero;
+  auto splitArg = [&](llvm::Optional<fir::CharBoxValue> arg,
+                      mlir::Value &buffer, mlir::Value &len) {
+    if (arg) {
+      buffer = arg->getBuffer();
+      len = arg->getLen();
+    } else {
+      if (!zero)
+        zero = builder.createIntegerConstant(loc, idxTy, 0);
+      buffer = zero;
+      len = zero;
+    }
+  };
+  mlir::Value dateBuffer;
+  mlir::Value dateLen;
+  splitArg(date, dateBuffer, dateLen);
+  mlir::Value timeBuffer;
+  mlir::Value timeLen;
+  splitArg(time, timeBuffer, timeLen);
+  mlir::Value zoneBuffer;
+  mlir::Value zoneLen;
+  splitArg(zone, zoneBuffer, zoneLen);
+
+  llvm::SmallVector<mlir::Value, 2> args{dateBuffer, timeBuffer, zoneBuffer,
```

Comment:
This won't be necessary when I get done with the Tilikum rework. All boxchar arguments will be deconstructed in the regular way.

Reply:
Just saw your big change :) I will update this to use your change once its merged

Reply:
Actually not possible to use here since the FIR signature of the runtime is already split (since the introspection that builds signature cannot group back the buffer and length into a boxchar).